### PR TITLE
Refactor multitask models to unify training and inference workflows

### PR DESCRIPTION
**Pull request by peipeijiang**
# Refactor multitask models to unify training and inference workflows

## Summary
This PR refactors the execution scripts for several multitask models (`APG_MMOE`, `APG_AITM`, `APG_SharedBottom`, `M3oE`) to support a unified entry point for both training and inference. It also promotes the inference data loader to the core framework to improve code reuse.

### 1. Core Framework
- **New Component**: Moved `DataFrameDataLoader` to `fuxictr/pytorch/dataloaders/dataframe_dataloader.py`.
- **Functionality**: This loader enables direct data loading from Pandas/Polars DataFrames (supporting Parquet and CSV formats) for inference, eliminating the need to regenerate TFRecords or h5 files for prediction tasks.

### 2. Unified `run_expid.py`
- Updated `run_expid.py` for the following models:
  - `APG_MMOE`
  - `APG_AITM`
  - `APG_SharedBottom`
  - `M3oE`
- **New Argument**: Added a `--mode` argument to switch between execution modes:
  - `train` (default): Performs standard model training and evaluation.
  - `inference`: Loads the saved `FeatureProcessor`, preprocesses raw input data (defined in `infer_data` config), and outputs predictions to a CSV file.

## Usage Example

**Training Mode:**
```bash
python run_expid.py --# Refactor multitask models to unify training and inference workflows

### CODE REVIEW

**Comment by linyuhui** [comment]
您好，您的邮件我已收到。如有需要，我会尽快给您回复。
